lib/src/elastic_client_impl.dart

```diff
@@ -265,18 +270,70 @@ class Client {
       results,
       suggestHits: suggestHits.isEmpty ? null : suggestHits,
       aggregations: aggResult.isEmpty ? null : aggResult,
+      scrollId: body['_scroll_id'] as String ?? null,
+    );
+  }
+
+  Future<SearchResult> scroll(String scrollId, Duration scroll) async {
+    final path = ['_search', 'scroll'];
+    final bodyMap = {'scroll_id': scrollId};
+    if (scroll != null) {
+      bodyMap['scroll'] = scroll.inSeconds.toString() + 's';
+    }
+    final rs = await _transport.send(Request('GET', path, bodyMap: bodyMap));
+    if (rs.statusCode != 200) {
+      throw Exception('Failed to search scroll');
+    }
+    final body = convert.json.decode(rs.body);
+    final hitsMap = body['hits'] ?? const {};
+    final hitsTotal = hitsMap['total'];
+    var totalCount = 0;
+    if (hitsTotal is int) {
+      totalCount = hitsTotal;
+    } else if (hitsTotal is Map) {
+      totalCount = (hitsTotal['value'] as int) ?? 0;
+    }
+    final hitsList = (hitsMap['hits'] as List).cast<Map>() ?? const <Map>[];
+    final results = hitsList
+        .map((Map map) => Doc(
```

Comment:
Let's reuse this from the other result processing code (e.g. refactor to a shared method or maybe into a constructor).

Reply:
I created _extractTotalCount and _extractDocList. Search method and scroll method use them.